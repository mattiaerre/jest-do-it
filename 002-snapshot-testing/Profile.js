import React from 'react';
import PropTypes from 'prop-types';

const Profile = (props) => {
  const handleOnClick = (e) => {
    e.preventDefault();
    props.doStuff();
  };

  return (
    <div className="profile-container">
      <h1>{props.name}</h1>
      <h2>{props.username}</h2>
      {props.bio && <p>{props.bio}</p>}
      <button onClick={handleOnClick}>Click me!</button>
    </div>
  );
};

Profile.defaultProps = {
  bio: null
};

Profile.propTypes = {
  name: PropTypes.string.isRequired,
  username: PropTypes.string.isRequired,
  bio: PropTypes.string,
  doStuff: PropTypes.func.isRequired
};

export default Profile;
