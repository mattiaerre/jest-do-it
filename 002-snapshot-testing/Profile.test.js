import React from 'react';
import renderer from 'react-test-renderer';
import Profile from './Profile';

test('<Profile />', () => {
  const props = {
    name: 'Mattia Richetto',
    username: '@mattiaerre',
    bio: 'Agile developer at OpenTable, marathon runner, sports fan and vegetarian living in San Francisco #FOB'
  };

  const component = renderer.create(
    <Profile {...props} />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
