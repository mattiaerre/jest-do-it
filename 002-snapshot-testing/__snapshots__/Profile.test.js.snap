// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> to match snapshot 1`] = `
<div
  className="profile-container"
>
  <h1>
    Mattia Richetto
  </h1>
  <h2>
    @mattiaerre
  </h2>
  <p>
    Agile developer at OpenTable, marathon runner, sports fan and vegan living in San Francisco #FOB
  </p>
  <button
    onClick={[Function]}
  >
    Click me!
  </button>
</div>
`;
